src/components/pages/schedule-page/schedule-event-edit-section/schedule-form/components/schedule-dropdown/TimeScheduleDropdown.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { FC, useEffect } from 'react';
+import dayjs from 'dayjs';
+import { useField } from 'formik';
+
+import Dropdown from '@/components/common/ui/form/dropdown';
+import { FormikDropdownProps } from '@/components/common/ui/form/with-formik/dropdown/FormikDropdown';
+
+import { getDateWithTimeSetted } from '../../utils/getDateWithTimeSetted';
+
+import * as styles from './ScheduleDropdown.styles';
+
+interface TimeScheduleDropdownProps extends FormikDropdownProps {
+  date: dayjs.Dayjs;
+}
+
+export const TimeScheduleDropdown: FC<TimeScheduleDropdownProps> = ({
+  clearOnUnmount,
+  name,
+  date,
+  ...props
+}) => {
+  const [{ value }, { touched, error }, { setValue, setTouched }] =
+    useField(name);
+
+  React.useEffect(() => {
+    setValue(getDateWithTimeSetted(date, value));
+  }, [date, value, setValue]);
```

Comment:
why do we need `setValue` on dependency?

Reply:
![image_2023-11-23_161747838](https://github.com/fictadvisor/fictadvisor-web/assets/131526240/b74b5145-fca2-4a99-8963-b257732e73b6)